Cargo.toml

```
@@ -31,12 +31,12 @@ rand = { version = "0.8", default-features = false, features = ["std_rng"] }
 rstest = "0.13"
 
 [features]
-default = ["fuel-asm/default", "fuel-crypto/default", "fuel-merkle?/default", "fuel-types/default", "std"]
-alloc = ["fuel-merkle", "fuel-types/alloc", "itertools/use_alloc", "serde/alloc"]
+default = ["fuel-asm/default", "fuel-crypto/default", "fuel-merkle/default", "fuel-types/default", "std"]
+alloc = ["fuel-types/alloc", "itertools/use_alloc", "serde/alloc"]
```

Just noticed we aren't properly leaving serde disabled when only alloc is enabled

```suggestion
alloc = ["fuel-types/alloc", "itertools/use_alloc", "serde?/alloc"]
```